src/main/java/org/apache/sysml/runtime/matrix/data/LibMatrixCUDA.java

```diff
@@ -1364,16 +1368,22 @@ private static void copyUpperToLowerTriangle(GPUContext gCtx, String instName, M
 	 * Examines sparsity and shapes and routes call to appropriate method
 	 * from cuBLAS or cuSparse
 	 * C = op(A) x op(B)
-	 * @param ec                    Current {@link ExecutionContext} instance
-	 * @param gCtx                  a valid {@link GPUContext}
-	 * @param instName              name of the invoking instruction to record{@link Statistics}.
-	 * @param left                  Matrix A
-	 * @param right                 Matrix B
-	 * @param outputName            Name of the output matrix C (in code generated after LOP layer)
-	 * @param isLeftTransposed      op for A, transposed or not
-	 * @param isRightTransposed     op for B, tranposed or not
-	 * @return	output of matrix multiply
+	 * <p>
+	 * Memory Requirements -
+	 * Both dense - inputs, output, no intermediate
+	 * Both sparse - inputs, output, no intermediate
```

Comment:
Are you sure that for all branches there are no intermediates? For example in our matrix multiplication library, we often use temporary memory for partial aggregates (w/o synchronization) in the number of threads.